src/oemof/solph/components/_offset_converter.py

```diff
@@ -282,12 +282,12 @@ def normed_offset_and_conversion_factors_from_coefficients(
                 c0, c1 = slope_offset_from_nonconvex_output(
                     flow.max[i], flow.min[i], eta_at_max, eta_at_min
                 )
-                slope += [c0]
-                offset += [c1]
+                slope.append(c0)
+                offset.append(c1)
 
             if max_len == 1:
-                slope = sequence(slope[0])
-                offset = sequence(offset[0])
+                slope = slope[0]
+                offset = offset[0]
 
```

Comment:
Thank you for jumping in here @p-snft. I was actually wrapping my head around this sequence stuff the first time I saw it when implementing the changes for the `OffsetConverter` and it took me quite a while to figure out what was going on. I see the practical point of the sequences, but I was wondering, if we could not just use scalars or `numpy` arrays instead in general. Because if you (by accident) provide a half length sequence (with respect to the timeindex) you would end up with some kind of nonsense anyway, right? So why not let people either pass scalars or correct-length lists or arrays?

Reply:
The problem is that we do not know the correct length before the Entity is added to the EnergySystem. An option would be to just let scalar stay scalars until we need to have arrays.